Library/Homebrew/cask/audit.rb

```diff
@@ -521,28 +521,32 @@ def audit_signing
             system_command("gktool", args: ["scan", path], print_stderr: false)
           when Artifact::Binary
             # Shell scripts cannot be signed, so we skip them
-            next if path.text_executable?
+            next false if path.text_executable?
 
             system_command("codesign",  args:         ["--verify", "-R=notarized", "--check-notarization", path],
                                         print_stderr: false)
           else
             add_error "Unknown artifact type: #{artifact.class}", location: url.location
           end
 
-          if result.success? && cask.deprecated? && cask.deprecation_reason == :unsigned
-            add_error "Cask is deprecated as unsigned but artifacts are signed!"
-          end
-
-          next if cask.deprecated? && cask.deprecation_reason == :unsigned
-
-          next if result.success?
+          next false if result.success?
+          next true if cask.deprecated? && cask.deprecation_reason == :unsigned
 
           add_error <<~EOS, location: url.location
             Signature verification failed:
             #{result.merged_output}
             macOS on ARM requires software to be signed.
             Please contact the upstream developer to let them know they should sign and notarize their software.
           EOS
+
+          true
+        end
+
+        if cask.deprecated? && cask.deprecation_reason == :unsigned && !any_signing_failure
+          add_error <<~EOS
+            Cask is deprecated as unsigned but all artifacts are signed!
+            Remove the deprecate/disable stanza or update the deprecate/disable reason.
+          EOS
         end
```

Comment:
Seems like you can return early here?
```suggestion
        return if any_signing_failure
        return unless cask.deprecated?
        return if cask.deprecation_reason != :unsigned

        add_error <<~EOS
          Cask is deprecated as unsigned but all artifacts are signed!
          Remove the deprecate/disable stanza or update the deprecate/disable reason.
        EOS
```